File: Perl/5.pl
#! /usr/bin/perl
use strict;
use warnings;

my $number = 2520;
my $found = 0;
my @div = (2..20);
my $index = 0;

while ($found != 1) {
#	print "Trying $number and $div[$index]\n";
	if ($number % $div[$index] != 0) {
#		print "Sorry but $number is divisible by $div[$index] so it failed\n";
		$number += 2;
		$index = 0;
	}
	elsif (($number % $div[$index] == 0) && ($index != 20)) {
		$index += 1;
	}
	else {
		$found = 1;
	}
}
print "The smallest number divisible by all numbers from 1-20 is $number\n";

File: CPP/5.cpp
/* Project Euler Problem 5
* Find the smallest number that is divisible by all numbers from 1-20
*/

#include <iostream>
using namespace std;

/* Function Prototypes */
int isDivisible(int n);


/* Main Program */
int main() {
	int test = 2520;
	int found = 0;
	while (found == 0) {
		test += 2;
		found = isDivisible(test);
	}
	cout << test << " is divisible by all numbers from 1-20." << endl;
	return 0;
}


/* Functions */
int isDivisible(int n) {
	int found = 1;
	for (int i = 2; i<= 20; i++) {
		if (n % i != 0) {
			found = 0;
			break;
		}
	}
	return found;
}
